### DISCUSSION TITLE
Lesson 3: gas fee

### DISCUSSION THREAD

**Discussion by nargeslein** [smartcontractkit/full-blockchain-solidity-course-py#436] [Q&A] [answered]
Hi everyone!
In Lesson 3 (at 2:54:25) a multiplication with 10000000000 is done in the function getPrice () in order to increase the lisibility of the price. Patrick says that it will save some gas if the multiplication is omitted. 
The function is a view function though. Is this a general recommendation for Solidity programming to avoid multiplications so that one avoids the risk of spending gas?
I am looking forward to your views (and especially Patrick's if he reads this).

![Screenshot 2021-11-24 224734](https://user-images.githubusercontent.com/33735545/143316938-e5467c20-afe0-4d9f-9451-1e7b738394ca.png)

**Comment by cromewar** [comment]
Hello @nargeslein I think is up to you, and how much you want to ensure the transactions on your contract will succeed or not.
In this case setting a little higher margin will ensure the users will pay the required Fee and the transaction will not get reverted, but of course the higher the amount is, the gas fees get higher.

It's also true that 10000000000 Wei is not significant to represent a real impact on the price so the gas difference would be despicable I think using this margins is more like a measure to ensure success rather than a bad practice.

Let's see if @PatrickAlphaC has another opinion on this.

best regards, Cromewar.

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question!

1. We multiplied by that amount to reach a sort of "standard". _Most_ prices and tokens are represented with 18 decimals, so we try to get the price to be represented as such as well.
2. Per the gas costs, you're right, `view` functions are free to call from off-chain! However... if a smart contract calls a view function in the transaction, it has to pay the gas fees to view it, otherwise it won't know what the price is!

View functions are only free if a simulator runs them, when we call a `view` function in remix, we actually only simulate calling it. When a smart contract calls it, it calls it with gas!

Does that make sense?

**Reply by cromewar** [reply]
This information is awesome, I forgot completely the 18 digits standard, you are awesome @PatrickAlphaC .

**Reply by nargeslein** [reply]
Many thanks @cromewar and @PatrickAlphaC, it's much clearer now.
@PatrickAlphaC, does point 2 also apply for pure function, e.g. they are free called off-chain but not if called within a transaction?

**Reply by PatrickAlphaC** [reply]
You got it! That's exactly correct
